optima/programs.py

```diff
@@ -603,29 +650,47 @@ def objectivecalc(progset=None, parset=None, year=None, ind=None, method='mape',
                 mismatch += thismismatch
             return mismatch
         
-        ## Do the actual calibration thingo
-        def cco2vec(): raise Exception('Not implemented')
+        # Store original values in case we need to go back to them
+        origcovout = dcp(self.covout)
+        origvals = dcp(self.cco2odict(t=year))
+        workingvals = dcp(origvals)
+        testarr = origvals[:] # Turn into array format
```

Comment:
is this the bit that's a 2xN vector when you want a 1x2N vector? should be easy to fix if so